app/gui/src/project-view/components/visualizations/toolbar.ts

```diff
@@ -13,21 +12,26 @@ export interface Button {
 export interface ActionButton extends Button {
   onClick: () => void
   disabled?: ToValue<boolean>
+  icon: Icon | URLString
 }
 
 export interface ToggleButton extends Button {
   toggle: Ref<boolean>
   disabled?: ToValue<boolean>
+  icon: Icon | URLString
 }
 
 export interface SelectionMenuOption extends Button {
   label?: string
+  icon?: Icon | URLString
 }
 
 export interface SelectionMenu {
   selected: Ref<string>
   title?: string
   options: Record<string, SelectionMenuOption>
+  isTextDropdown?: boolean
+  heading?: string
 }
```

Comment:
Instead of the `isTextDropdown` field, let's introduce a separate type:
```ts
export interface TextSelectionMenu {
  type: 'textSelectionMenu'
  selected: Ref<string>
  title?: string
  options: Record<string, TextSelectionMenuOption>
  heading?: string
}
```

---

The reason for the `type` field approach is probably not apparent--it fits into my plan for the `ToolbarItem` types (as we add more and their definitions get more complex):

Eventually, every type will have a `type` field that must contain a specific value if present; however, the field will be optional for any type that can be recognized by its fields, to allow concise toolbar definitions.

So far, every type has had distinctive enough fields that we haven't needed to make any `type` fields required (and I haven't got around to defining optional `type` fields). The new `TextSelectionMenu` has similar fields to the normal `SelectionMenu`, so its `type` field is not optional.

Reply:
ok! Thanks, I wasn't sure if it was best to adapt the current dropdown or make a new one so I'll separate it now!

Reply:
I should have made those changes and got them working now!